### PR TITLE
feat: support listen http2 with plaintext

### PR DESCRIPTION
**Pull request by hnlq715**
### What this PR does / why we need it:

Details can be found in #3496.

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [X] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [X] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/cli/ops.lua [hunk 1]
```diff
@@ -238,10 +238,18 @@ Please modify "admin_key" in conf/config.yaml .
...
     end
 
-    --support multiple ports listen, compatible with the original style
+    -- support multiple ports listen, compatible with the original style
     if type(yaml_conf.apisix.node_listen) == "number" then
-        local node_listen = {yaml_conf.apisix.node_listen}
+        local node_listen = {{port = yaml_conf.apisix.node_listen}}
         yaml_conf.apisix.node_listen = node_listen
+    elseif type(yaml_conf.apisix.node_listen) == "table" then
+        if type(yaml_conf.apisix.node_listen[1]) == "number" then
```
Comment (tokers) [inline apisix/cli/ops.lua:246]: Cannot capture your mean for this if.

File: t/core/config-default.t [hunk 2]
```diff
@@ -91,3 +91,53 @@ apisix:
 GET /t
 --- response_body
 admin_key: null
+
+
+
+=== TEST 5: support listen multiple ports with array
```
Comment (spacewander) [inline t/core/config-default.t:97]: Need to add tests in t/grpc-proxy-test.sh.
And need to update the doc/grpc-proxy.md

File: conf/config.yaml [hunk 3]
```diff
@@ -31,6 +31,12 @@
 #
 # If the configured environment variable can't be found, an error will be thrown.
 apisix:
+  node_listen:
+    - port: 9080                     # APISIX listening port
```
Comment (spacewander) [inline conf/config.yaml:35]: We should provide the configuration separately in:
https://github.com/apache/apisix/blob/26a4ed4ad491f140fe441df281365c44723cae20/.travis/linux_openresty_common_runner.sh#L125

File: apisix/cli/ops.lua (hunk 1 above)
Comment (spacewander) [reply apisix/cli/ops.lua:246]: I think we should check every item, what if the user mixes the integer and the table?

Comment (hnlq715) [reply apisix/cli/ops.lua:246]: Only support port with number parameter?

File: t/core/config-default.t (hunk 2 above)
Comment (hnlq715) [reply t/core/config-default.t:97]: done

File: doc/grpc-proxy.md [hunk 4]
```diff
@@ -36,7 +36,8 @@ gRPC client -> APISIX -> gRPC/gRPCS server
 Here's an example, to proxying gRPC service by specified route:
 
 * attention: the `scheme` of the route's upstream must be `grpc` or `grpcs`.
-* attention: APISIX use TLS‑encrypted HTTP/2 to expose gRPC service, so need to [config SSL certificate](https.md)
+* attention: APISIX use TLS‑encrypted HTTP/2 to expose gRPC service (default port 9443), so need to [config SSL certificate](https.md)
+* attention: APISIX also support to expose gRPC service with plaintext HTTP/2 (default port 9081), which do not need to config SSL certificate, usually used to proxy gRPC service in intranet environment
```
Comment (spacewander) [inline doc/grpc-proxy.md:40]: Need a configuration to show how we can do it.

File: conf/config.yaml [hunk 5]
```diff
@@ -31,6 +31,12 @@
...
+      enable_http2: false            # Disable to support HTTP proxy, nginx currently do not support HTTP and HTTP/2 with plaintext on the same port
+                                     # Details can be found in https://trac.nginx.org/nginx/ticket/816
+    - port: 9081                     # APISIX listening port for HTTP/2 with plaintext
```
Comment (spacewander) [inline conf/config.yaml:38]: Don't open a port for HTTP/2 by default.

Comment (hnlq715) [reply conf/config.yaml:38]: done

File: doc/grpc-proxy.md (hunk 4 above)
Comment (hnlq715) [reply doc/grpc-proxy.md]: done

File: conf/config.yaml (hunk 3 above)
Comment (hnlq715) [reply conf/config.yaml:35]: done

**Comment by spacewander** [comment]
Need to resolve the conflict, so that CI can start.

File: doc/grpc-proxy.md [hunk 6]
```diff
@@ -37,6 +37,7 @@ Here's an example, to proxying gRPC service by specified route:
 
 * attention: the `scheme` of the route's upstream must be `grpc` or `grpcs`.
 * attention: APISIX use TLS‑encrypted HTTP/2 to expose gRPC service, so need to [config SSL certificate](https.md)
+* attention: APISIX also support to expose gRPC service with plaintext HTTP/2, which do not need to config SSL certificate, usually used to proxy gRPC service in intranet environment
```
Comment (tokers) [inline doc/grpc-proxy.md:40]: which do not need to config SSL certificate => which doesn't rely on ALPN/NPN extensions in TLS/SSL protocol.

File: apisix/cli/ops.lua [hunk 7]
```diff
@@ -238,9 +238,19 @@ Please modify "admin_key" in conf/config.yaml .
...
-    --support multiple ports listen, compatible with the original style
+    -- support multiple ports listen, compatible with the original style
     if type(yaml_conf.apisix.node_listen) == "number" then
-        local node_listen = {yaml_conf.apisix.node_listen}
+        local node_listen = {{port = yaml_conf.apisix.node_listen}}
+        yaml_conf.apisix.node_listen = node_listen
+    elseif type(yaml_conf.apisix.node_listen) == "table" then
+        local node_listen = {}
+        for index, value in ipairs(yaml_conf.apisix.node_listen) do
+            if type(value) == "number" then
```
Comment (tokers) [inline apisix/cli/ops.lua:248]: Is the number type really needed? I think the following way is enough.

```yaml
node_listen:
  - port: 9080
  - port: 9180
    enable_http2: true
```

Comment (spacewander) [reply apisix/cli/ops.lua:248]: We need to be backward compatibility.

**Review by spacewander — APPROVED** [review]

File: doc/grpc-proxy.md (hunk 6 above)
Comment (hnlq715) [reply doc/grpc-proxy.md:40]: Maybe this is a little hard to understand for user...? Maybe it's enough to mention in doc that this feature is supported, and show how to configure it?

File: apisix/cli/ops.lua (hunk 7 above)
Comment (tokers) [reply apisix/cli/ops.lua:248]: I mean the element type inside `node_listen`, not the `node_listen` itself.

File: doc/grpc-proxy.md (hunk 6 above)
Comment (tokers) [reply doc/grpc-proxy.md:40]: What about "which doesn't rely on TLS/SSL"?

File: apisix/cli/ops.lua (hunk 7 above)
Comment (hnlq715) [reply apisix/cli/ops.lua:248]: We can also be compatible with this format:

```yaml
node_listen:
    - 9080
    - 9081
```

File: doc/grpc-proxy.md (hunk 6 above)
Comment (hnlq715) [reply doc/grpc-proxy.md:40]: Alright

**Review by tokers — APPROVED** [review]

**Pull request merged**